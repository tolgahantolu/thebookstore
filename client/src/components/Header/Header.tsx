import React from 'react';

const Header: React.FC = () => {
  return (
    <div
      data-testid="header-image"
      className="mt-10 relative w-full h-[440px] bg-header-pattern-1 bg-center"
    >
      <div
        data-testid="header-text"
        className="absolute top-1/2 left-4 sm:left-10 md:left-20 bg-theme-white px-4 sm:px-6 py-2 sm:py-3 w-72 h-auto xs:w-auto"
      >
        <q className="text-2xl sm:text-3xl md:text-4xl font-semibold block">
          Today a reader, tomorrow a leader.
        </q>
        <span className="sm:text-sm md:text-base font-normal xs:font-medium italic block xs:text-end mt-3 mr-5">
          - Margaret Fuller
        </span>
      </div>
    </div>
  );
};

export default Header;
