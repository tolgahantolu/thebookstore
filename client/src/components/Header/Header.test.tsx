import { render, screen } from '@testing-library/react';
import Header from './Header';

describe('Header Component', () => {
  beforeEach(() => {
    render(<Header />);
  });

  it('should render image correctly', () => {
    expect(screen.getByTestId('header-image')).toMatchSnapshot();
  });

  it('should render header text correctly', () => {
    expect(screen.getByTestId('header-text')).toHaveTextContent(
      /Today a reader, tomorrow a leader./i,
    );
  });
});
