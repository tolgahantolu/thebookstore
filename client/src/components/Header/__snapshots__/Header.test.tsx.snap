// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render image correctly 1`] = `
<div
  class="mt-10 relative w-full h-[440px] bg-header-pattern-1 bg-center"
  data-testid="header-image"
>
  <div
    class="absolute top-1/2 left-20 bg-theme-white px-6 py-3"
    data-testid="header-text"
  >
    <q
      class="text-4xl font-semibold block"
    >
      Today a reader, tomorrow a leader.
    </q>
    <span
      class="font-medium italic block text-end mt-3 mr-5"
    >
      - Margaret Fuller
    </span>
  </div>
</div>
`;
